### PR TITLE
Percent-encode GraphQL repository.url

### LINKED ISSUE #15618: References panel fetches highlighted file without %-decoding repo from URI
Repro: https://k8s.sgdev.org/sourcegraph.visualstudio.com/Test%20Repo/-/blob/Test%20Folder/Test%20File.cs#L7:20&tab=references

image

The highlight request includes the `%` from the URI of the repo (interestingly not of the file):

image

We should add an extension integration test for this.

### PR DESCRIPTION
**Pull request by felixfbecker**
I had narrowed down the problem to specific resolvers so here's my best shot to fix #15618.

What I am unsure about:
- Is this the right location for the utility function? I don't know our code organization in the backend well.
- Is the unit test good?
- Should there be some kind of integration test for the GraphQL fields? (I don't know how to add those)

### CODE REVIEW

File: cmd/frontend/graphqlbackend/repository.go [hunk 1]
```diff
@@ -243,10 +243,11 @@ func (r *RepositoryResolver) UpdatedAt() *DateTime {
 }
 
 func (r *RepositoryResolver) URL() string {
+	url := "/" + pathEscapeExceptSlashes(string(r.repo.Name))
```
Comment (unknwon) [inline cmd/frontend/graphqlbackend/repository.go:246]: It feels like there is no point to keep both `escapeRevspecForURL` and `pathEscapeExceptSlashes` as all the former does is to call the latter.

It is probably better to just rename the `escapeRevspecForURL` to ``escapePathForURL`. WDYT?

**Comment by unknwon** [comment]
Thanks, merging!

**Comment by felixfbecker** [comment]
@unknwon what do you think about an integration test for this to prevent regressions? The unit test tests that the utility function, but nothing tests that it's actually used, so I'm a bit worried about this issue popping up again in the future

**Comment by unknwon** [comment]
> @unknwon what do you think about an integration test for this to prevent regressions? The unit test tests that the utility function, but nothing tests that it's actually used, so I'm a bit worried about this issue popping up again in the future

@felixfbecker SGTM. Would any of GitHub, Bitbucket Server or AWS Code Commit supports spaces in repository name? 🤔 If not, might be a bit troublesome to add.

**Comment by felixfbecker** [comment]
GitHub definitely not. The others I don't know, but I imagine not. But since Azure DevOps is supported by Sourcegraph through "Other", it would be good if the integration tests could test that, no?

**Comment by unknwon** [comment]
> GitHub definitely not. The others I don't know, but I imagine not. But since Azure DevOps is supported by Sourcegraph through "Other", it would be good if the integration tests could test that, no?

@felixfbecker Agree with the point of having it tested.

Do you have a working example of the "Other" external service for Azure DevOps, doesn't seem we have docs for it.

**Comment by felixfbecker** [comment]
No, I had to figure it out myself too 🙈
You can check the config that worked on k8s.sgdev.org though.

**Comment by unknwon** [comment]
> No, I had to figure it out myself too 🙈
> You can check the config that worked on k8s.sgdev.org though.

The one on the k8s.sgdev.org does look like we would need exactly. Filed https://github.com/sourcegraph/sourcegraph/issues/15905.

**Pull request merged**